### ISSUE TITLE
Keras activation not plotted if input is dropout layer.

### ISSUE THREAD

**Issue by ananroha** [keras-team/keras#7801]
I have trained the following model for Cifar10. I am using Theano as backend. You can read the summary here:-
-------------------------------------------------------------------------------------------------------------------
model = Sequential()

model.add(Conv2D(32, (3, 3), padding='same',name="act1",
                 input_shape=x_train.shape[1:],activation='relu'))

model.add(Conv2D(32, (3, 3),name='act2',activation='relu'))

model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Conv2D(64, (3, 3), padding='same',name='act3',activation='relu'))

model.add(Conv2D(64, (3, 3),name='act4',activation='relu'))

model.add(MaxPooling2D(pool_size=(2, 2)))
model.add(Dropout(0.25))

model.add(Flatten())
model.add(Dense(512))
model.add(Activation('relu'))
model.add(Dropout(0.5))
model.add(Dense(num_classes))
model.add(Activation('softmax'))
-------------------------------------------------------------------------------------------------------------------
Now you can see that before 'act3' layer there is a dropout layer. Now I am trying to create a partial graph by using the theano.function() and I am compiling it from Input tensor to the network to the desired layer.
Now this step works till the 'act2' layer. After that, for 'act3' layer the input tensor changes to if().0 (The output tensor of dropout) which is correct. But the problem is that the graph can not be compiled when I give the input as the 1st layer of the network. This is very strange because dropout is inside the network and hence the partial graph should be compilable from main input to 'act3' layer.
I have compiled the graph from 'dropout_1' to 'act3' and it works perfectly. It is very strange the graph from main input to 'act3' can not be compiled. But all the layers before 'act3' can be compiled from main input. I dont thing that is logical, as dropout is a part of network and its input depends on the previous layers. Could you please explain why this is so. Am I making an error in interpretation or is this a bug in keras. I also find this strange because the model itself is compilable and can be trained correctly.

Note when I say compiled from x to y, I mean:-
theano.function([x],y)
